tests/ssg_test_suite/oscap.py

```diff
@@ -44,29 +61,63 @@ def run_profile(domain_ip,
                                                                    stage))
     verbose_path = os.path.join(LogHelper.LOG_DIR, '{0}-{1}'.format(profile,
                                                                     stage))
+    arf_path = os.path.join(LogHelper.LOG_DIR,
+                            '{0}-initial-arf.xml'.format(profile))
+
     formatting['report'] = LogHelper.find_name(report_path, '.html')
     verbose_path = LogHelper.find_name(verbose_path, '.verbose.log')
 
-    command = shlex.split(('oscap-ssh root@{domain_ip} 22 xccdf eval '
-                           '--benchmark-id {benchmark_id} '
-                           '--profile {profile} '
-                           '--progress --oval-results '
-                           '--report {report} '
-                           '--verbose DEVEL '
-                           '{rem} '
-                           '{datastream}').format(**formatting))
-    logging.debug('Running ' + ' '.join(command))
-    success = True
-    try:
-        with open(verbose_path, 'w') as verbose_file:
-            output = subprocess.check_output(command, stderr=verbose_file)
-    except subprocess.CalledProcessError, e:
-        # non-zero exit code
-        if e.returncode != 2:
-            success = False
+    if stage is 'initial' and ansible:
+        formatting['arf'] = "--results-arf {}".format(arf_path)
+    else:
+        formatting['arf'] = ""
+
+    if stage is 'remediation' and ansible:
+        formatting['arf'] = arf_path
+        formatting['ansible_template'] = _ANSIBLE_TEMPLATE
+        formatting['playbook'] = os.path.join(LogHelper.LOG_DIR,
+                                              '{0}.yml'.format(profile))
+        command = shlex.split(('oscap xccdf generate fix '
+                               '--benchmark-id {benchmark_id} '
+                               '--profile {profile} '
+                               '--template {ansible_template} '
+                               '--output {playbook} '
+                               '{arf}').format(**formatting))
+        if run_cmd(command, verbose_path)[0] is not 0:
+            return False
+        # Updates ansible playbok hosts so we can run it later against
+        # the tested domain_ip.
+        f = fileinput.FileInput(formatting['playbook'], inplace=True)
```

Comment:
can't you just do open(..).readlines()? seems overkill to use the multi file input streaming for this

Reply:
ah I see, you redir stdout to it

use `with` guard to make it less dangerous, we don't want stdout to be redirected forever especially if any errors are printed to it.

I suspect there probably is a cleaner solution to this whole thing.

Reply:
I rewritten this so it is more clear and also used `with` when opening files.